Content.Shared/_Harmony/ItemCurse/SharedItemCurseSystem.cs

```diff
@@ -0,0 +1,239 @@
+            TryMarkItem(ent, markItem.Value);
+            return;
+        }
+
+        // Finger snap emote because it's Cool.
+        Snap(args.Performer);
+        CurseItem(ent.Comp.MarkedEntity.Value, ent.Comp);
+        args.Handled = true;
+    }
+
+    private void CurseItem(Entity<CurseMarkerComponent?> ent, ItemCurseComponent comp)
+    {
+        if (!Resolve(ent.Owner, ref ent.Comp, false))
+            return;
+
+        if (!TryComp<InstantActionComponent>(ent.Comp.MarkedByAction, out var instantAction))
+            return;
+
+        var actionOwner = instantAction.AttachedEntity;
+
+        if (actionOwner == null)
+            return;
+
+        // Un-embed embeddable projectiles
+        if (TryComp<EmbeddableProjectileComponent>(ent, out var projectile))
+            _proj.EmbedDetach(ent, projectile, actionOwner.Value);
+
+        // Check if the cursed item is in any container.
+        // If it is, check each outer container until either finding a body to target or running out of containers.
+        // The first body found will be targeted by the direct shocking effect.
+        var containerEnt = ent;
+        while (_containerSystem.TryGetContainingContainer((containerEnt, null, null), out var nextContainer))
+        {
+            if (HasComp<BodyComponent>(nextContainer.Owner))    // Is checking for the body comp the right way to do this? I hope so.
+            {
+                ShockHolder(nextContainer.Owner, ent, comp);
+                break;
+            }
+            containerEnt = nextContainer.Owner;
+        }
+
+        // If the entity is in a container, place it into the world instead
+        if (_containerSystem.IsEntityInContainer(ent))
+        {
+            _transform.SetCoordinates(ent, Transform(ent.Owner).Coordinates);
+            _transform.AttachToGridOrMap(ent);
+        }
+
+        // Give it a good chuck
+        _throwing.TryThrow(ent, _random.NextVector2(), baseThrowSpeed: comp.FlingStrength);
+        CreateLightning(ent, comp);
+    }
+
+    private void OnCurseMarkerShutdown(Entity<CurseMarkerComponent> ent, ref ComponentShutdown args)
+    {
+        TryUnmarkItem(ent);
+    }
+
+    private void TryMarkItem(Entity<ItemCurseComponent> ent, EntityUid item)
+    {
+        if (!TryComp<InstantActionComponent>(ent, out var instantAction))
+            return;
+
+        var actionOwner = instantAction.AttachedEntity;
+
+        if (actionOwner == null)
+            return;
+
+        AddToPvsOverride(item, actionOwner.Value);
+
+        var marker = AddComp<CurseMarkerComponent>(item);
+        ent.Comp.MarkedEntity = item;
+        Dirty(ent);
+
+        marker.MarkedByAction = ent.Owner;
+
+        UpdateActionAppearance(ent);
+        Dirty(item, marker);
+    }
+
+    private void TryUnmarkItem(EntityUid item)
+    {
+        if (!TryComp<CurseMarkerComponent>(item, out var marker))
+            return;
+
```

Comment:
You can just make the function directly take in an entity with a curse marker component
```suggestion
    private void TryUnmarkItem(Entity<CurseMarkerComponent> item)
    {
```